restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/fcm/FcmConfigurationClient.java

```diff
@@ -0,0 +1,528 @@
+     */
+    private FcmBinding createOrUpdateBinding(FcmBinding binding, String bindingSid){
+        RCLogger.v(TAG, "createBinding method started");
+        FcmBinding fcmBinding = null;
+        try{
+            String url = "https://" + pushDomain + "/" + this.pushPath + "/bindings";
+            if (bindingSid != null){
+                url = "https://" + pushDomain + "/" + this.pushPath + "/bindings/" + bindingSid;
+            }
+            RCLogger.v(TAG, "calling url: " + url);
+            HttpURLConnection connection = createUrlRequestWithUrl(url);
+            connection.setRequestMethod("POST");
+            if (bindingSid != null){
+                connection.setRequestMethod("PUT");
+            }
+            connection.setRequestProperty("Content-Type", "application/json");
+            String outputString =
+                    "{\n" +
+                            "  \"Identity\": \"" + binding.getIdentity() + "\",\n" +
+                            "  \"ApplicationSid\": \"" + binding.getApplicationSid() + "\",\n" +
+                            "  \"BindingType\": \"" + binding.getBindingType() + "\",\n" +
+                            "  \"Address\": \"" + binding.getAddress() + "\"\n" +
+                            "}";
+            OutputStream outputStream = new BufferedOutputStream(connection.getOutputStream());
+            outputStream.write(outputString.getBytes());
+            outputStream.close();
+            int responseCode = connection.getResponseCode();
+
+            if (responseCode == HttpURLConnection.HTTP_OK){
+                fcmBinding = getBindingFromConnection(connection, binding.getApplicationSid());
+            }
+
+        }catch (Exception ex){
+            RCLogger.e(TAG, ex.toString());
+        }
+        RCLogger.v(TAG, "createBinding method ended; returning fcmBinding");
+        return fcmBinding;
+    }
+
+
+
+
+    //-------------------------Helper methods -----------------------------//
+
+    private FcmApplication getApplicationFromConnection(HttpURLConnection connection, String applicationName) throws IOException, JSONException{
+        String jsonString = getJsonFromStream(connection.getInputStream());
+        if (jsonString.length() > 0) {
+            Object json = new JSONTokener(jsonString).nextValue();
+            if (json instanceof  JSONArray){
+                JSONArray inputJson = new JSONArray(jsonString);
+                for (int i = 0; i < inputJson.length(); i++){
+                    JSONObject client = inputJson.getJSONObject(i);
+                    return extractFromJsonObject(client, applicationName);
+                }
+            } else {
+                JSONObject client = new JSONObject(jsonString);
+                return extractFromJsonObject(client, applicationName);
+            }
+
+        }
+        return null;
+    }
+
+    private FcmApplication extractFromJsonObject(JSONObject client, String applicationName) throws IOException, JSONException{
```

Comment:
Should we maybe call this extractFromJsonObjectApplication instead of extractFromJsonObject for consistency?

Reply:
Yeah, agree.
Changed.